### PR TITLE
Quick fix for composer with xdebug active

### PR DESCRIPTION
**Pull request by vasilegoian**
Instead of running composer directly, we create a separate binary for
composer and a shell script that will disable the xdebug, run the
composer binary with all the arguments passed to the script and then
reenable the xdebug.

### CODE REVIEW

**Comment by cytopia** [comment]
@vasilegoian thanks for the PR. Is composer running with xdebug causing any problems?

**Comment by vasilegoian** [comment]
@cytopia yes. For some reason, when xdebug is enabled (at least on PHP7.2) composer won't run at all. No output (let it run for a few hours), no errors, nothing. Disabling xdebug instantly fix the issue. Tried it on Docker for Mac version 2.0.0.0-mac81.

**Comment by cytopia** [comment]
Can you try the following command (`-n` makes it not load any php ini files):
```
php -n /usr/local/bin/composer <your composer arguments here>
```

**Comment by cytopia** [comment]
Alternatively I could even try to replace the shebang of `/usr/local/bin/composer` itself:

```diff
- #!/usr/bin/env php
+ #!/usr/local/bin/php -n
```

**Comment by vasilegoian** [comment]
That works, but causes other issues when specific extensions are required by composer and the `-n` flag would make php ignore all of them. Composer has a flag `--ignore-platform-reqs` that ignores the  missing extensions, but this could cause other issues by installing the required packages without the correct requirements being met.

The problem being specifically xdebug, I think disabling it while composer runs and reenabling it afterwards is the best approach. The only side effect is that if you kill a composer action and don't run another one afterwards, xdebug will remain disabled.

**Comment by cytopia** [comment]
> The only side effect is that if you kill a composer action and don't run another one afterwards, xdebug will remain disabled.

I was also thinking the same.
I might have to split up the ini directives into two parts: One for php-cli and another one for php-fpm.

Btw, in order for your changes to be reflected in the Dockerfiles itself, you need to run `make generate` (which also requires `ansible` to be installed).

File: build/ansible/group_vars/all.yml [hunk 1]
```diff
@@ -345,7 +345,18 @@ software_available:
...
+        { \
+        echo '#!/bin/bash'; \
+        echo 'if [ -f /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini ]; then'; \
+        echo '	sudo mv /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini.bk'; \
+        echo 'fi'; \
+        echo 'composer-binary "$@"'; \
+        echo 'if [ -f /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini.bk ]; then'; \
+        echo '  sudo mv /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini.bk /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini'; \
+        echo 'fi'; \
+        } > /usr/local/bin/composer \
```
Comment (cytopia) [inline build/ansible/group_vars/all.yml]: This part is actually overwriting the `composer` binary. Composer is installed in `/usr/local/bin/composer` (not `composer-binary`). You can tell by the Travis CI checks, that all consequent steps with composer fail.

**Comment by cytopia** [comment]
As I've already said, this is not the desired solution as it causes the race-condition to accidentally disable xdebug. What I can suggest as a quick fix for your case is to have a script for yourself which does exactly this in a fixed way (see above comment that you currently overwrite the `compose` binary`).

See the following documentation link for how to add custom autostart scripts: https://devilbox.readthedocs.io/en/latest/autostart/custom-scripts-per-php-version.html

Additionally you are right about the lack for contributing and I will have to improve this repository with some information about all the required software and steps.

**Comment by vasilegoian** [comment]
Thanks! I'll try this method instead. It indeed sounds better suited.

**Pull request closed without merging**